tests/test_text.py

```diff
@@ -123,6 +123,13 @@ def test_copy():
     assert text == test_copy
     assert text is not test_copy
 
+def test_format():
```

Comment:
This needs more tests.

Check plain strings, strings with a single substitution, no styles, several styles, overlapping styles, etc. Try to cover as many edge cases as you can.